### PR TITLE
[json-rpc] Seperate an reusable json-rpc client crate

### PR DESCRIPTION
**Pull request by thefallentree**
create an reusable json-rpc client crate that could be used for consuming json-rpc service.

No behavior change, just moving stuff around.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -45,6 +46,7 @@ members = [
     "execution/executor-types",
     "execution/executor-utils",
     "json-rpc",
+    "json-rpc",
```
Comment (msmouse) [inline Cargo.toml:49]: ?

Comment (thefallentree) [reply Cargo.toml]: deleted.

File: client/json-rpc/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+version = "0.1.0"
+authors = ["Libra Association <opensource@libra.org>"]
+description = "Libra JSONRPC client"
+repository = "https://github.com/libra/libra"
+homepage = "https://libra.org"
+license = "Apache-2.0"
+publish = false
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (JoshLind) [inline client/json-rpc/Cargo.toml:12]: Silly question: is this comment really needed? I've never seen this in a .toml file before

File: client/json-rpc/src/client.rs [hunk 3]
```diff
@@ -120,11 +120,20 @@ pub struct JsonRpcAsyncClient {
 }
 
 impl JsonRpcAsyncClient {
+    /// Deprecated, use `new_from_url`
     pub fn new(client: Client, host: &str, port: u16) -> Self {
```
Comment (JoshLind) [inline client/json-rpc/src/client.rs:124]: Silly question number 2 😄What's the reason behind deprecating this call? Also if you want to deprecate it, can we not just remove the method and update all the call sites?

Comment (mimoo) [reply client/json-rpc/src/client.rs:124]: you also have this thing I believe: https://doc.rust-lang.org/edition-guide/rust-2018/the-compiler/an-attribute-for-deprecation.html

File: client/json-rpc/Cargo.toml (hunk 2 above)
Comment (thefallentree) [reply client/json-rpc/Cargo.toml:12]: it is just part of the template that generated by clion, I can remove it.

**Comment by thefallentree** [comment]
going to move the blocking JsonRPCClient in CLI to this crate as well

File: testsuite/cluster-test/src/tx_emitter.rs [hunk 4]
```diff
@@ -306,9 +304,10 @@ impl TxEmitter {
 
     fn make_client(&self, instance: &Instance) -> JsonRpcAsyncClient {
         JsonRpcAsyncClient::new(
-            self.http_client.clone(),
```
Comment (andll) [inline testsuite/cluster-test/src/tx_emitter.rs:309]: We actually want to reuse same HttpClient for each JsonRpcAsyncClient, why change it to creating new HttpClient every time?

File: client/json-rpc/src/client.rs (hunk 3 above)
Comment (thefallentree) [reply client/json-rpc/src/client.rs]: done, deleted all call site that uses the old one

File: testsuite/cluster-test/src/tx_emitter.rs (hunk 4 above)
Comment (thefallentree) [reply testsuite/cluster-test/src/tx_emitter.rs:309]: yeah, you are right, fixed.

File: testsuite/cluster-test/src/tx_emitter.rs [hunk 5]
```diff
@@ -305,10 +306,12 @@ impl TxEmitter {
     }
 
     fn make_client(&self, instance: &Instance) -> JsonRpcAsyncClient {
-        JsonRpcAsyncClient::new(
+        JsonRpcAsyncClient::new_with_client(
             self.http_client.clone(),
-            instance.ip(),
-            instance.ac_port() as u16,
+            Url::from_str(
+                format!("http://{}:{}", instance.ip(), instance.ac_port() as u16).as_str(),
```
Comment (andll) [inline testsuite/cluster-test/src/tx_emitter.rs:312]: very small nit: you probably no longer need `as u16` in this context

Comment (thefallentree) [reply testsuite/cluster-test/src/tx_emitter.rs:312]: fixed.

**Comment by andll** [comment]
@bors-libra delegate+
Looks good!

**Pull request closed without merging**